openclsim/core.py

```diff
@@ -1194,6 +1200,61 @@ def __init__(
             else:
                 print("No known optimization method selected")
 
+    def check_optimal_filling_Roadmap(self, loader, unloader, origin, destination):
+        orig = shapely.geometry.asShape(origin.geometry)
+        dest = shapely.geometry.asShape(destination.geometry)
+        start = (orig.x, orig.y)
+        stop = (dest.x, dest.y)
+        prods = []
+        tQQ = []
+
+        for i in range(len(self.loadfactors)):
+            # Departure time sailing empty
+            t0 = datetime.datetime.fromtimestamp(self.env.now).strftime(
+                "%d/%m/%Y %H:%M:%S"
+            )
+
+            # Duration of sailing empty
+            _, TT, _ = self.optimization_func(
+                stop, start, t0, self.env.Roadmap.vship[0, -1], self.env.Roadmap
+            )
+
+            # Duration of sailing empty + loading
+            duration_dredging = self.loading(
+                None, None, self.container.capacity * self.loadfactors[i]
+            )
+
+            TTT = self.env.now + (TT[-1] - TT[0]) + duration_dredging
+            duration_sailing_empty = TT[-1] - TT[0]
+
+            # Departure time sailing full
+            t0 = datetime.datetime.fromtimestamp(TTT).strftime("%d/%m/%Y %H:%M:%S")
+
+            # Duration of sailing full
+            _, TT, _ = self.optimization_func(
+                start, stop, t0, self.env.Roadmap.vship[i, -1], self.env.Roadmap
+            )
+
+            # Duration of sailing empty + loading + sailing full + unloading
+            duration_unloading = self.unloading(
+                None, None, self.container.capacity * self.loadfactors[i]
+            )
+            TTT += (TT[-1] - TT[0]) + duration_unloading
```

Comment:
This doesn't seem necessary for further computations.

Reply:
This part is needed for the calculation of the optimal load factor